src/Microsoft.AspNetCore.Authentication.Core/AuthenticationService.cs

```diff
@@ -183,13 +201,61 @@ public AuthenticationService(IAuthenticationSchemeProvider schemes, IAuthenticat
                 }
             }
 
-            var handler = await Handlers.GetHandlerAsync(context, scheme) as IAuthenticationSignOutHandler;
+            var handler = await Handlers.GetHandlerAsync(context, scheme);
             if (handler == null)
             {
-                throw new InvalidOperationException($"No IAuthenticationSignOutHandler is configured to handle sign out for the scheme: {scheme}");
+                var schemes = await GetAllSignOutSchemeNames();
+                throw new InvalidOperationException(
+                    $"No authentication handler is registered for the scheme '{scheme}'. The registered sign-out schemes are: {schemes}");
+            }
+
+            var signOutHandler = handler as IAuthenticationSignOutHandler;
+            if (signOutHandler == null)
+            {
+                var schemes = await GetAllSignOutSchemeNames();
+                throw new InvalidOperationException(
+                    $"The authentication handler registered for scheme '{scheme}' is '{handler.GetType().Name}' which cannot be used for {nameof(SignOutAsync)}. "
+                    + $"The registered sign-out schemes are: {schemes}");
+            }
+
+            await signOutHandler.SignOutAsync(properties);
+        }
+
+        private async Task<string> GetAllSchemeNames()
+        {
+            var schemes = string.Join(", ", (await Schemes.GetAllSchemesAsync()).Select(sch => sch.Name));
+
+            if (string.IsNullOrEmpty(schemes))
+            {
+                return "[none]";
```

Comment:
Naw, I think the message should be different instead of saying none.

Reply:
"There are no registered authentication schemes."?
Then what?

Reply:
"You need to add your authentication scheme to the IAuthenticationBuilder returned by AddAuthentication()"

Reply:
Or "Did you forget to call AddAuthentication().AddScheme()?"

Reply:
There's only so much we can cram in here. How about a fwlink to the auth docs instead?

Reply:
The EF team is notorious for good error message like this. It's *super* valuable.